### PR TITLE
#227 Add unit tests

### PR DESCRIPTION
**Pull request by sgerogia**

### CODE REVIEW

File: api/chains/chains_test.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			// if we have a populated Chain store, add it
+			if !cmp.Equal(tt.dataStruct, cns.Chain{}) {
+				err := testingCtx.CnsDB.AddChain(tt.dataStruct)
+				require.NoError(t, err)
+			}
+
+			// act
+			resp, err := http.Get(fmt.Sprintf(chainEndpointUrl, testingCtx.Cfg.ListenAddr, tt.chainName))
+			defer func() { _ = resp.Body.Close() }()
+			require.NoError(t, err)
```
Comment (DeshErBojhaa) [inline api/chains/chains_test.go]: nit picking, but order of these 3 lines should be,
```go
resp, err := http.Get(fmt.Sprintf(chainEndpointUrl, testingCtx.Cfg.ListenAddr, tt.chainName))
require.NoError(t, err)
defer func() { _ = resp.Body.Close() }()
````
Same for other tests.

File: api/chains/chains_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
+package chains_test
```
Comment (DeshErBojhaa) [inline api/chains/chains_test.go:1]: 👏🏼 for external test.

File: api/test_utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+		CnsDB:  cns,
+	}
+}
+
+// TruncateDB Empties the DB of data.
+// Only use in tests executed sequentially
+func TruncateDB(ctx *TestingCtx, t *testing.T) {
+	// FIXME: Using DB util from another service
+	_, err := ctx.CnsDB.Instance.DB.Exec("TRUNCATE cns.chains")
+	assert.NoError(t, err)
```
Comment (DeshErBojhaa) [inline api/test_utils/utils.go]: Question: since we're using `require` almost everywhere, I wonder why we're using assert here?

**Review by DeshErBojhaa — CHANGES_REQUESTED** [review]
Almost there!

File: api/test_utils/utils.go (hunk 3 above)
Comment (sgerogia) [reply api/test_utils/utils.go]: My bad, maybe copy-paste. Replaced.

File: api/chains/chains_test.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+				body, err := ioutil.ReadAll(resp.Body)
+				require.NoError(t, err)
+
+				respStruct := chains.ChainResponse{}
+				err = json.Unmarshal(body, &respStruct)
+				require.NoError(t, err)
+
+				require.Equal(t, tt.expectedHttpCode, resp.StatusCode)
+				require.Equal(t, tt.dataStruct, respStruct.Chain)
+			}
```
Comment (DeshErBojhaa) [inline api/chains/chains_test.go]: One suggestion, some people like it some don't, your call if you consider it or not.
```go
if !tt.success {
    require.Equal(t, tt.expectedHttpCode, resp.StatusCode)
    continue
}

body, err := ioutil.ReadAll(resp.Body)
require.NoError(t, err)

respStruct := chains.ChainResponse{}
err = json.Unmarshal(body, &respStruct)
require.NoError(t, err)

require.Equal(t, tt.expectedHttpCode, resp.StatusCode)
require.Equal(t, tt.dataStruct, respStruct.Chain)

```
Keeps nesting 1 level less.
However, if you opt for this approach, please change all the following occurrences.

**Review by DeshErBojhaa — APPROVED** [review]
LGTM! With one comment!

File: api/chains/chains_test.go (hunk 4 above)
Comment (sahith-narahari) [reply api/chains/chains_test.go]: I agree, this is more readable and better to avoid nesting

**Review by sahith-narahari — CHANGES_REQUESTED** [review]

File: api/chains/chains_test.go (hunk 4 above)
Comment (sgerogia) [reply api/chains/chains_test.go]: Fixed with `return`, as inside a method rather than a loop

**Review by sahith-narahari — APPROVED** [review]

**Pull request merged**